### ISSUE TITLE
prevent-abbreviations: whitelist has no effect on file name

### ISSUE THREAD

**Issue by mrmlnc** [sindresorhus/eslint-plugin-unicorn#567]
### Environment

* OS Version: Windows 10
* Node.js Version: `13.7.0`
* ESLint: `6.8.0`  
* `eslint-plugin-unicorn`: `16.1.1`

### Actual behavior

```
eslint-plugin-unicorn-repro/some.spec.e2e.js
1:1  error  Please rename the filename `./some.spec.e2e.js`.
Suggested names are: `some.spec.error2error.js`, `some.spec.error2event.js`, `some.spec.event2error.js`, ... (1 more omitted).
A more descriptive name will do too
unicorn/prevent-abbreviations

✖ 1 problem (1 error, 0 warnings)
```

### Expected behavior

There are no errors, because `e2e` is a valid word.

Also this works for the variable name, but not for file name.

### Steps to reproduce

1. `npm i eslint eslint-plugin-unicorn`
2. Create `.eslintrc.json` file
3. Create `some.spec.e2e.js` file with content: `var e2e = 1;`
4. `npx eslint .`

### Config

```json
{
    "plugins": ["unicorn"],
    "rules": {
        "unicorn/prevent-abbreviations": [
            "error",
            {
                "whitelist": {
                    "e2e": true
                }
            }
        ]
    }
}
```

**Comment by mrmlnc** [comment]
This behavior is related to the line:

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/875c471be0ce5023a9b9af9f8cf62cd59efe634c/rules/prevent-abbreviations.js#L305-L306

```js
// words
[
  'some', '.',
  'spec', '.',
  'e', '2', 'e'
]
```

**Comment by fisker** [comment]
It was me implement this word split method, but I have no idea how to fix this case. 

If we split word includes the number, `err2` will get pass, since we are replacing `err` not `err2`.

If we ignore `e2e` like replace `some.spec.e2e` to `some.spec._placeholder_`, works for this case  but, if we choice ignore `e` , ` err` will get pass again.

**Comment by fisker** [comment]
`var e2e = 1;` is ignored because the whole `e2e` can be ignore before splitting into words. for filename, you can also get pass, if your file name is `e2e.js`. And if you name `var is_e2e =true` you will get the same failed result as `some.spec.e2e.js`, they are the  same logic.

**Comment by mrmlnc** [comment]
Sorry for the delay. Unfortunately, I don't see the right solution here.

It seems that the safest way is to have a separate whitelist of words for file names and cut them out of the file names.

Most likely, this issue describes the edge case.

**Comment by fisker** [comment]
Came up a solution, check #662 .
